### PR TITLE
fix NPE in getExpectedType

### LINKED ISSUE #2306: NoSuchMethodError: The getter 'type' was called on null.Receiver: nullTried calling: type for at getExpectedType(unnecessary_null_checks.dart:108)
This is from internal exception reporting - seen 16 times by 1 user (sample report 827fe7446c156a5b):

```
NoSuchMethodError: The getter 'type' was called on null.Receiver: nullTried calling: type
at Object.noSuchMethod(dart:core)
at getExpectedType(unnecessary_null_checks.dart:108)
at _Visitor.visitPostfixExpression(null_check_on_nullable_type_parameter.dart:81)
at PostfixExpressionImpl.accept(ast.dart:8104)
at LinterVisitor._runSubscriptions(linter_visitor.dart:755)
at LinterVisitor.visitPostfixExpression(linter_visitor.dart:539)
at PostfixExpressionImpl.accept(ast.dart:8104)
at NodeListImpl.accept(ast.dart:7553)
at ArgumentListImpl.visitChildren(ast.dart:360)
at RecursiveAstVisitor.visitArgumentList(visitor.dart:629)
at LinterVisitor.visitArgumentList(linter_visitor.dart:36)
at ArgumentListImpl.accept(ast.dart:356)
at MethodInvocationImpl.visitChildren(ast.dart:7136)
at RecursiveAstVisitor.visitMethodInvocation(visitor.dart:1073)
at LinterVisitor.visitMethodInvocation(linter_visitor.dart:480)
at MethodInvocationImpl.accept(ast.dart:7129)
at InterpolationExpressionImpl.visitChildren(ast.dart:6260)
at RecursiveAstVisitor.visitInterpolationExpression(visitor.dart:1013)
at LinterVisitor.visitInterpolationExpression(linter_visitor.dart:420)
at InterpolationExpressionImpl.accept(ast.dart:6256)
at NodeListImpl.accept(ast.dart:7553)
at StringInterpolationImpl.visitChildren(ast.dart:9239)
at RecursiveAstVisitor.visitStringInterpolation(visitor.dart:1218)
at LinterVisitor.visitStringInterpolation(linter_visitor.dart:625)
at StringInterpolationImpl.accept(ast.dart:9235)
at NodeListImpl.accept(ast.dart:7553)
at ArgumentListImpl.visitChildren(ast.dart:360)
at RecursiveAstVisitor.visitArgumentList(visitor.dart:629)
at LinterVisitor.visitArgumentList(linter_visitor.dart:36)
at ArgumentListImpl.accept(ast.dart:356)
at MethodInvocationImpl.visitChildren(ast.dart:7136)
at RecursiveAstVisitor.visitMethodInvocation(visitor.dart:1073)
at LinterVisitor.visitMethodInvocation(linter_visitor.dart:480)
at MethodInvocationImpl.accept(ast.dart:7129)
at ExpressionStatementImpl.visitChildren(ast.dart:3655)
at RecursiveAstVisitor.visitExpressionStatement(visitor.dart:839)
at LinterVisitor.visitExpressionStatement(linter_visitor.dart:246)
at ExpressionStatementImpl.accept(ast.dart:3651)
at NodeListImpl.accept(ast.dart:7553)
at BlockImpl.visitChildren(ast.dart:1074)
at RecursiveAstVisitor.visitBlock(visitor.dart:671)
at LinterVisitor.visitBlock(linter_visitor.dart:78)
at BlockImpl.accept(ast.dart:1070)
at IfStatementImpl.visitChildren(ast.dart:5652)
at RecursiveAstVisitor.visitIfStatement(visitor.dart:977)
at LinterVisitor.visitIfStatement(linter_visitor.dart:384)
at IfStatementImpl.accept(ast.dart:5647)
at NodeListImpl.accept(ast.dart:7553)
at BlockImpl.visitChildren(ast.dart:1074)
at RecursiveAstVisitor.visitBlock(visitor.dart:671)
at LinterVisitor.visitBlock(linter_visitor.dart:78)
at BlockImpl.accept(ast.dart:1070)
at BlockFunctionBodyImpl.visitChildren(ast.dart:1029)
at RecursiveAstVisitor.visitBlockFunctionBody(visitor.dart:677)
at LinterVisitor.visitBlockFunctionBody(linter_visitor.dart:84)
at BlockFunctionBodyImpl.accept(ast.dart:1025)
at MethodDeclarationImpl.visitChildren(ast.dart:6985)
at RecursiveAstVisitor.visitMethodDeclaration(visitor.dart:1067)
at LinterVisitor.visitMethodDeclaration(linter_visitor.dart:474)
at MethodDeclarationImpl.accept(ast.dart:6976)
at NodeListImpl.accept(ast.dart:7553)
at ClassDeclarationImpl.visitChildren(ast.dart:1552)
at RecursiveAstVisitor.visitClassDeclaration(visitor.dart:707)
at LinterVisitor.visitClassDeclaration(linter_visitor.dart:114)
at ClassDeclarationImpl.accept(ast.dart:1522)
at NodeListImpl.accept(ast.dart:7553)
at CompilationUnitImpl.visitChildren(ast.dart:2120)
at RecursiveAstVisitor.visitCompilationUnit(visitor.dart:731)
at LinterVisitor.visitCompilationUnit(linter_visitor.dart:138)
at CompilationUnitImpl.accept(ast.dart:2113)
at LibraryAnalyzer._computeLints(library_analyzer.dart:366)
at LibraryAnalyzer.analyzeSync(library_analyzer.dart:176)
at LibraryAnalyzer.analyze(library_analyzer.dart:107)
at AnalysisDriver._computeAnalysisResult2.<fn>(driver.dart:1317)
at PerformanceLog.run(performance_logger.dart:32)
at AnalysisDriver._computeAnalysisResult2(driver.dart:1294)
at AnalysisDriver._computeAnalysisResult.<fn>(driver.dart:1247)
at runZoned(dart:async)
at NullSafetyUnderstandingFlag.enableNullSafetyTypes(null_safety_understanding_flag.dart:42)
at AnalysisDriver._computeAnalysisResult(driver.dart:1246)
at AnalysisDriver.performWork(driver.dart:1089)
at AnalysisDriverScheduler._run(driver.dart:2013)
at _completeOnAsyncReturn(dart:async)
at Monitor.signal(status.dart)
```

### PR DESCRIPTION
**Pull request by a14n**
Fixes #2306

### CODE REVIEW

File: lib/src/rules/unnecessary_null_checks.dart [hunk 1]
```diff
@@ -104,7 +104,7 @@ DartType getExpectedType(PostfixExpression node) {
     parent = parent.parent;
   }
   if (parent is ArgumentList) {
-    return (realNode as Expression).staticParameterElement.type;
+    return (realNode as Expression).staticParameterElement?.type;
```
Comment (pq) [inline lib/src/rules/unnecessary_null_checks.dart:107]: It would be great to have a test but if a repro is elusive not a big deal.

**Review by pq — APPROVED** [review]
Thanks!

File: lib/src/rules/unnecessary_null_checks.dart (hunk 1 above)
Comment (a14n) [reply lib/src/rules/unnecessary_null_checks.dart:107]: I don't know how to repro :/

Comment (bwilkerson) [reply lib/src/rules/unnecessary_null_checks.dart:107]: It isn't clear to me which lint rule is being run here, but `staticParameterElement` should only be `null` if the argument doesn't correspond to a parameter. My first guess would be something like:
```
int f() => 0;
var x = f(2);
```
where the `2` is some expression that would have caused this code to be reached. If it needs to be a named argument then use a name that isn't defined for `f`.

**Comment by pq** [comment]
Given we have a roll pending, I'll land this now.  @a14n: maybe backfill with a test if you cook one up?

**Pull request merged**